catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/ProjectListFragment.java

```diff
@@ -162,11 +186,29 @@ private void importProject(Intent data) {
 			return;
 		}
 
+		if (!data.getData().getScheme().equals("file")) {
+			throw new RuntimeException("importProject has to be called with a file uri. (not a content uri");
+		}
+
 		try {
-			File projectZip = StorageOperations
-					.copyUriToDir(getContext().getContentResolver(), data.getData(), CACHE_DIR, CACHED_PROJECT_ZIP_FILE_NAME);
-			new ProjectUnzipAndImportTask(this)
-					.execute(projectZip);
+			File cacheFile = new File(CACHE_DIR, CACHED_PROJECT_ZIP_FILE_NAME);
```

Comment:
just out of curiosity: how does the file at `CACHED_PROJECT_ZIP_FILE_NAME` become a directory? and is it then a directory that ends with ".zip"? 

from what I've seen below you just directly copy the project if `src` is a directory and just create a file if it's a zip. am I missing something here?